tests/core/util/test_streamable.py

```diff
@@ -252,28 +243,28 @@ class ConvertPrimitiveFailures:
 def test_convert_primitive_failures(input_dict: Dict[str, Any], error: Any) -> None:
 
     with pytest.raises(error):
-        dataclass_from_dict(ConvertPrimitiveFailures, input_dict)
+        streamable_from_dict(ConvertPrimitiveFailures, input_dict)
 
 
 @pytest.mark.parametrize(
     "test_class, input_dict, error",
     [
-        [TestDataclassFromDict1, {"a": "asdf", "b": "2", "c": G1Element()}, TypeError],
-        [TestDataclassFromDict1, {"a": 1, "b": "2"}, KeyError],
-        [TestDataclassFromDict1, {"a": 1, "b": "2", "c": "asd"}, TypeError],
-        [TestDataclassFromDict1, {"a": 1, "b": "2", "c": "00" * G1Element.SIZE}, TypeError],
-        [TestDataclassFromDict1, {"a": [], "b": "2", "c": G1Element()}, TypeError],
-        [TestDataclassFromDict1, {"a": {}, "b": "2", "c": G1Element()}, TypeError],
-        [TestDataclassFromDict2, {"a": "asdf", "b": 1.2345, "c": 1.2345}, TypeError],
-        [TestDataclassFromDict2, {"a": 1.2345, "b": {"a": 1, "b": "2"}, "c": 1.2345}, TypeError],
-        [TestDataclassFromDict2, {"a": {"a": 1, "b": "2", "c": G1Element()}, "b": {"a": 1, "b": "2"}}, KeyError],
-        [TestDataclassFromDict2, {"a": {"a": 1, "b": "2"}, "b": {"a": 1, "b": "2"}, "c": 1.2345}, KeyError],
+        [StreamableFromDict1, {"a": "asdf", "b": "2", "c": G1Element()}, TypeError],
+        [StreamableFromDict1, {"a": 1, "b": "2"}, KeyError],
+        [StreamableFromDict1, {"a": 1, "b": "2", "c": "asd"}, TypeError],
+        [StreamableFromDict1, {"a": 1, "b": "2", "c": "00" * G1Element.SIZE}, TypeError],
+        [StreamableFromDict1, {"a": [], "b": "2", "c": G1Element()}, TypeError],
+        [StreamableFromDict1, {"a": {}, "b": "2", "c": G1Element()}, TypeError],
+        [StreamableFromDict2, {"a": "asdf", "b": 12345, "c": 12345}, TypeError],
+        [StreamableFromDict2, {"a": 12345, "b": {"a": 1, "b": "2"}, "c": 12345}, TypeError],
+        [StreamableFromDict2, {"a": {"a": 1, "b": "2", "c": G1Element()}, "b": {"a": 1, "b": "2"}}, KeyError],
+        [StreamableFromDict2, {"a": {"a": 1, "b": "2"}, "b": {"a": 1, "b": "2"}, "c": 12345}, KeyError],
     ],
 )
-def test_dataclass_from_dict_failures(test_class: Type[Any], input_dict: Dict[str, Any], error: Any) -> None:
+def test_streamable_from_dict_failures(test_class: Type[_T_Streamable], input_dict: Dict[str, Any], error: Any) -> None:
```

Comment:
I think this would work?  If so then the import can be removed as well.

```suggestion
def test_streamable_from_dict_failures(test_class: Type[Streamable], input_dict: Dict[str, Any], error: Any) -> None:
```

Reply:
Right, adjusted and rebased on latest `main`